pallets/rc-migrator/src/accounts.rs

```diff
@@ -715,8 +755,63 @@ impl<T: Config> AccountsMigrator<T> {
 					id.to_ss58check(),
 					rc_reserved
 				);
-				RcAccounts::<T>::insert(&id, AccountState::Part { reserved: rc_reserved });
+				RcAccounts::<T>::insert(
+					&id,
+					// one consumer reference of reserved balance.
+					AccountState::Part { free: rc_ed, reserved: rc_reserved, consumers: 1 },
+				);
+			}
+		}
+
+		// every key owner has to preserve ED on the RC with a consumer reference.
+
+		let rc_ed = <T as Config>::Currency::minimum_balance();
+		let ah_ed = T::AhExistentialDeposit::get();
+		for (who, _keys) in pallet_session::NextKeys::<T>::iter() {
+			weight += T::DbWeight::get().reads(1);
+			if let Some(mut state) = RcAccounts::<T>::get(&who) {
+				match state {
+					AccountState::Part { ref mut consumers, .. } => {
+						log::debug!(
+							target: LOG_TARGET,
+							"Increase session pallet consumer reference for account state: {:?}",
+							who.to_ss58check(),
+						);
+						*consumers += 1;
```

Comment:
Just to point out the obvious: Accounts cannot appear multiple times in `NextKeys` since it is a map.